src/charm.py

```diff
@@ -0,0 +1,130 @@
+        dummy_http_server = DummyHTTPServer(self, self.DUMMY_SERVER_PORT)
+        dummy_http_server.start_server()
+
+    def _on_pebble_ready(self, event: PebbleReadyEvent):
+        """Checks whether all conditions to start Prometheus Configurer are met and, if yes,
+        triggers start of the prometheus-configurer service.
+        """
+        if not self._relations_created:
+            event.defer()
+            return
+        if self._container.can_connect():
+            self._start_prometheus_configurer()
+        else:
+            self.unit.status = WaitingStatus(
+                f"Waiting for {self._container_name} container to be ready"
+            )
+            event.defer()
+
+    def _start_prometheus_configurer(self):
+        """Starts Prometheus Configurer service."""
+        plan = self._container.get_plan()
+        layer = self._prometheus_configurer_layer
+        if plan.services != layer.services:
+            self.unit.status = MaintenanceStatus(
+                f"Configuring pebble layer for {self._service_name}"
+            )
+            self._container.add_layer(self._container_name, layer, combine=True)
+            self._container.restart(self._service_name)
+            logger.info(f"Restarted container {self._service_name}")
+            self.unit.status = ActiveStatus()
+
+    def _push_alert_rules(self, _):
+        """Pushes alert rules to Prometheus through the relation data bag."""
+        topology = JujuTopology.from_charm(self)
+        alert_rules = AlertRules(topology=topology)
+        alert_rules.add_path(self.RULES_DIR, recursive=True)
+        alert_rules_as_dict = alert_rules.as_dict()
+        alert_rules_content = (
+            alert_rules_as_dict if alert_rules_as_dict["groups"][0]["rules"] else []
+        )
+        if alert_rules_as_dict:
+            prometheus_relation = self.model.get_relation("prometheus")
+            prometheus_relation.data[self.app]["alert_rules"] = json.dumps(alert_rules_content)  # type: ignore[union-attr]  # noqa: E501
+
+    @property
+    def _prometheus_configurer_layer(self) -> Layer:
+        """Construct the pebble layer for Prometheus configurer.
+
+        Returns:
+            a Pebble layer specification for the Prometheus configurer workload container.
+        """
+        return Layer(
+            {
+                "summary": "Prometheus Configurer layer",
+                "description": "Pebble layer configuration for Prometheus Configurer",
+                "services": {
+                    self._service_name: {
+                        "override": "replace",
+                        "startup": "enabled",
+                        "command": "prometheus_configurer "
+                        f"-port={str(self._prometheus_configurer_port)} "
+                        f"-rules-dir={self.RULES_DIR}/ "
+                        f"-prometheusURL=127.0.0.1:{self.DUMMY_SERVER_PORT} "
+                        "-multitenant-label=networkID "
```

Comment:
Was `networkID` a magma specific thing? we may want to pass it has a config or relation